dm/loader/checkpoint.go

```diff
@@ -101,7 +102,18 @@ type RemoteCheckPoint struct {
 }
 
 func newRemoteCheckPoint(tctx *tcontext.Context, cfg *config.SubTaskConfig, id string) (CheckPoint, error) {
-	db, dbConns, err := createConns(tctx, cfg, cfg.Name, cfg.SourceID, 1)
+	var err error
+	var db *conn.BaseDB
+	var dbConns []*DBConn
+
+	rollbackHolder := fr.NewRollbackHolder("loader")
+	defer func() {
+		if err != nil {
+			rollbackHolder.RollbackReverseOrder()
+		}
+	}()
+
+	db, dbConns, err = createConns(tctx, cfg, cfg.Name, cfg.SourceID, 1)
 	if err != nil {
 		return nil, err
```

Comment:
seems if err != nil  we have not add add rollbcak func into`rollbackHolder` yes , so there is no rollbak in defer🤔, can we add a ut to test the rowback func really works?

Reply:
> ~seems if err != nil we have not add add rollbcak func into`rollbackHolder` yes , so there is no rollbak in defer🤔, can we add a ut to test the rowback func really works?~
> 
> oh i see it's trying to close db when cp.prepare meet error

bingo